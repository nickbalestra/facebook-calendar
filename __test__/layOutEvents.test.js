const { Column } = require("../src/components/events/layOutEvents");

test("insert a confliting event in a new column", () => {
  const firstColumn = new Column();
  firstColumn.insert({ start: 0, end: 400 });
  expect(firstColumn.nextColumn).toBeUndefined();
  firstColumn.insert({ start: 0, end: 400 });
  expect(firstColumn.nextColumn).toBeInstanceOf(Column);
});

test("track conflicting groups length", () => {
  const firstColumn = new Column();
  [
    { start: 0, end: 400 },
    { start: 100, end: 300 },
    { start: 100, end: 200 },
    { start: 200, end: 300 }
  ].forEach(event => firstColumn.insert(event));
  firstColumn.traverseAll(event => {
    expect(event._intersectionGroup.length).toBe(3);
  });
});

test("group events by conflicts", () => {
  const firstColumn = new Column();
  [
    { start: 0, end: 400 },
    { start: 100, end: 300 },
    { start: 100, end: 200 },
    { start: 200, end: 300 },
    { start: 400, end: 600 },
    { start: 400, end: 500 }
  ].forEach(event => firstColumn.insert(event));
  firstColumn.traverseAll(event => {
    expect(event).toMatchSnapshot();
  });
});
