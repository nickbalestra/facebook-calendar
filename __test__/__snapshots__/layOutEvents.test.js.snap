// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`group events by conflicts 1`] = `
Object {
  "_intersectionGroup": Object {
    "end": 400,
    "length": 3,
  },
  "end": 400,
  "offset": 0,
  "start": 0,
}
`;

exports[`group events by conflicts 2`] = `
Object {
  "_intersectionGroup": Object {
    "end": 600,
    "length": 2,
  },
  "end": 600,
  "offset": 0,
  "start": 400,
}
`;

exports[`group events by conflicts 3`] = `
Object {
  "_intersectionGroup": Object {
    "end": 400,
    "length": 3,
  },
  "end": 300,
  "offset": 1,
  "start": 100,
}
`;

exports[`group events by conflicts 4`] = `
Object {
  "_intersectionGroup": Object {
    "end": 600,
    "length": 2,
  },
  "end": 500,
  "offset": 1,
  "start": 400,
}
`;

exports[`group events by conflicts 5`] = `
Object {
  "_intersectionGroup": Object {
    "end": 400,
    "length": 3,
  },
  "end": 200,
  "offset": 2,
  "start": 100,
}
`;

exports[`group events by conflicts 6`] = `
Object {
  "_intersectionGroup": Object {
    "end": 400,
    "length": 3,
  },
  "end": 300,
  "offset": 2,
  "start": 200,
}
`;
